shell/src/main/java/alluxio/cli/fs/command/DecommissionWorkerCommand.java

```diff
@@ -0,0 +1,127 @@
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Decommission a specific worker, the worker complete decommissioning when
+ * all the replicas from them are replicated to other workers. Decommissioned
+ * worker are not automatically shutdown and are not chosen for writing new
+ * replicas.
+ */
+public final class DecommissionWorkerCommand extends AbstractFileSystemCommand {
+
+  private static final int DEFAULT_TIMEOUT = 10 * Constants.MINUTE_MS;
+  private static final Option TIMEOUT_OPTION =
+      Option.builder("t")
+          .longOpt("timeout")
+          .argName("timeout in milliseconds")
+          .numberOfArgs(1)
+          .desc("Timeout in milliseconds for decommissioning a"
+              + "single worker, default: " + DEFAULT_TIMEOUT)
+          .required(false)
+          .build();
+
+  private static final Option HOST_OPTION =
+      Option.builder("host")
+          .longOpt("host")
+          .required(true)  // Host option is mandatory.
+          .hasArg(true)
+          .numberOfArgs(1)
+          .argName("host")
+          .desc("A worker host name, which is mandatory.")
+          .build();
+
+  private static final Option FORCE_OPTION =
+      Option.builder("f")
+          .required(false)
+          .hasArg(false)
+          .desc("Force to decommission a worker.")
+          .build();
+
+  /**
+   * Constructs a new instance to decommission the given worker from Alluxio.
+   * @param fsContext the filesystem of Alluxio
+   */
+  public DecommissionWorkerCommand(FileSystemContext fsContext) {
+    super(fsContext);
+  }
+
+  @Override
+  public int run(CommandLine cl) throws AlluxioException, IOException {
+    int timeoutMS = FileSystemShellUtils.getIntArg(cl, TIMEOUT_OPTION, DEFAULT_TIMEOUT);
+    String workerHost = cl.getOptionValue(HOST_OPTION.getLongOpt());
+
+    DecommissionWorkerPOptions options =
+            DecommissionWorkerPOptions.newBuilder()
+                .setTimeOut(timeoutMS).setForced(cl.hasOption("f")).build();
+
+    List<BlockWorkerInfo> cachedWorkers = mFsContext.getCachedWorkers();
+
+    for (BlockWorkerInfo blockWorkerInfo :cachedWorkers)  {
+      if (Objects.equals(blockWorkerInfo.getNetAddress().getHost(), workerHost))  {
+        try {
+          long start = System.currentTimeMillis();
+          mFileSystem.decommissionWorker(blockWorkerInfo.getNetAddress(), options);
+          long duration = System.currentTimeMillis() - start;
+          System.out.printf("Decommission worker %s success, spend: %dms%n",
+              workerHost, duration);
+        } catch (InterruptedException ie) {
+          System.out.println(ie);
+        }
+        return 0;
+      }
+    }
+
+    System.out.println("Target worker is not found in Alluxio, please input another name.");
+    return 0;
```

Comment:
how about
```suggestion
    BlockWorkerInfo tempBlockWorkerInfo = NULL;
    for (BlockWorkerInfo blockWorkerInfo : cachedWorkers) {
      if (Objects.equals(blockWorkerInfo.getNetAddress().getHost(), workerHost)) {
        tempBlockWorkerInfo = blockWorkerInfo;
        break;
      }
    }

    if (tempBlockWorkerInfo != NULL)  {
      try {
        long start = System.currentTimeMillis();
        mFileSystem.decommissionWorker(blockWorkerInfo.getNetAddress(), options);
        long duration = System.currentTimeMillis() - start;
        System.out.printf("Decommission worker %s success, spend: %dms%n",
            workerHost, duration);
      } catch (InterruptedException ie) {
        System.out.println(ie);
      }
    }
    else {
      System.out.println("Target worker is not found in Alluxio, please input another name.");
    }

    return 0;
```

Reply:
The old code is like the code I wrote four months ago......

Reply:
@YichuanSun My origin implementation is far different from your design, so I just copied your codes, can we let this pr be the first step to the decommission feature?

Reply:
> @YichuanSun My origin implementation is far different from your design, so I just copied your codes, can we let this pr be the first step to the decommission feature?

OK, but we need more exprienced alluxio engineer to review this PR. @dbw9580 Could you please take a look when you have time? thanks!